File: leetcode/problems/py/left-and-right-sum-differences.py
# by @codeAbinash
# Time : O(n)
# Space : O(n)

import functools


class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        r_sum = functools.reduce(lambda a, b: a + b, nums)
        l_sum = 0
        result = []
        for i in range(len(nums)):
            r_sum -= nums[i]
            result.append(abs(l_sum - r_sum))
            l_sum += nums[i]
        return result

File: leetcode/problems/cpp/left-and-right-sum-differences.cpp
// by @codeAbinash
// Time : O(n)
// Space : O(n)

#include <vector>
using namespace std;

class Solution {
 public:
   vector<int> leftRightDifference(vector<int>& nums) {
      int n = nums.size();
      vector<int> ans(n);

      int r_sum = 0, l_sum = 0;
      for (int i = 0; i < n; i++) r_sum += nums[i];

      for (int i = 0; i < n; i++) {
         r_sum -= nums[i];
         ans[i] = abs(r_sum - l_sum);
         l_sum += nums[i];
      }

      return ans;
   }
};